src/components/Hping3/Hping3.tsx

```diff
@@ -0,0 +1,267 @@
+import { Button, Stack, Switch, TextInput, NumberInput, NativeSelect } from "@mantine/core";
+import { useForm } from "@mantine/form";
+import { useCallback, useState, useEffect } from "react";
+import { CommandHelper } from "../../utils/CommandHelper";
+import ConsoleWrapper from "../ConsoleWrapper/ConsoleWrapper";
+import { RenderComponent } from "../UserGuide/UserGuide";
+import { SaveOutputToTextFile_v2 } from "../SaveOutputToFile/SaveOutputToTextFile"; //v2
+import { LoadingOverlayAndCancelButtonPkexec } from "../OverlayAndCancelButton/OverlayAndCancelButton";
+import { checkAllCommandsAvailability } from "../../utils/CommandAvailability";
+import InstallationModal from "../InstallationModal/InstallationModal";
+
+/**
+ * Represents the form values for the hping3 component.
+ */
+interface FormValuesType {
+    ipAddress: string;
+    minPort: string;
+    maxPort: string;
+    portNumber: string;
+    toolOptions: string;
+    packetCount: string;
```

Comment:
```suggestion
    additionalParams: string; // Allows for more advanced commands to be executed
```

Reply:
Will park this for later - I think hping3 is definitely a tool that can be expanded on in later trimesters.